### PR TITLE
Task/14070 Link Card - Pandemic Radar

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Provides an easy access to the RKI pandemic radar.
Adds new Link Cards.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14048
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14070
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14083

### CODE REVIEW

**Review by CV113 — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -2041,8 +2041,11 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 /* - General Card Texts */
 
 "Statistics_Card_From_Nationwide" = "Bundesweit";
+
 "Statistics_Card_From_CWA" = "über die Corona-Warn-App";
 
+"Statistics_Card_LinkCard_buttonTitle" = "Externen Link öffnen  ";
+
```
Comment (30mar) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2048]: is this extra space intentional ?

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -2055,6 +2058,17 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
...
 
+/* - Pandemic Radar */
+
+"Statistics_Card_PandemicRadar_title" = "Pandemieradar";
+
+"Statistics_Card_PandemicRadar_subtitle" = "des Robert-Koch-Instituts";
+
+"Statistics_Card_PandemicRadar_description" = "Entdecken Sie weitere Statistiken zur Pandemie";
+
+"Statistics_Card_PandemicRadar_buttonTitle" = "Externen Link öffnen  ";
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2069]: Please remove extra space after öffnen

File: src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+extension NSAttributedString {
+	var attachementImage: UIImage? {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift:8]:
```suggestion
	var attachmentImage: UIImage? {
```

File: src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		var image: UIImage?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift:9]: Missing private? 

```suggestion
		private var image: UIImage?
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCard.swift [hunk 5]
```diff
@@ -0,0 +1,9 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+enum HomeLinkCard: Int32, CaseIterable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCard.swift:7]: Perhaps for future, we need the enums for all the cards as we have discussed

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+	}
+	
+	// MARK: - Private
+	
+	@IBOutlet private weak var containerStackView: UIStackView!
+	@IBOutlet private weak var middleStackView: UIStackView!
+	@IBOutlet private weak var titleLabel: ENALabel!
+	@IBOutlet private weak var subtitleLabel: ENALabel!
+	@IBOutlet private weak var descriptionLabel: StackViewLabel!
+	@IBOutlet private weak var assetImageView: UIImageView!
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:120]: Image is an asset

```suggestion
	@IBOutlet private weak var imageView: UIImageView!
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+			self.deleteButton.alpha = enabled ? 1 : 0
+		} completion: { _ in
+			self.deleteButton.alpha = enabled ? 1 : 0
+		}
+	}
+	
+	// MARK: - Private
+	
+	@IBOutlet private weak var containerStackView: UIStackView!
+	@IBOutlet private weak var middleStackView: UIStackView!
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:116]: Can we more explicit, what is middleStackView?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+		case .pandemicRadar:
+			setupPandemicRadar(for: linkCard)
+		case .none:
+			Log.info("Link card ID \(linkCard.header.cardID) is not supported", log: .ui)
+		}
+	}
+	
+	// MARK: - Internal
+	
+	@OpenCombine.Published private(set) var title: String?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:23]: They all are private, better to put them under private mark

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+	@OpenCombine.Published private(set) var subtitle: String?
+	@OpenCombine.Published private(set) var description: String?
+	@OpenCombine.Published private(set) var asset: UIImage?
+	@OpenCombine.Published private(set) var buttonTitle: NSAttributedString?
+	@OpenCombine.Published private(set) var buttonURL: URL?
+	
+	@OpenCombine.Published private(set) var titleAccessibilityIdentifier: String?
+	@OpenCombine.Published private(set) var subtitleAccessibilityIdentifier: String?
+	@OpenCombine.Published private(set) var descriptionAccessibilityIdentifier: String?
+	@OpenCombine.Published private(set) var assetAccessibilityIdentifier: String?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:33]:
```suggestion
	@OpenCombine.Published private(set) var imageAccessibilityIdentifier: String?
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+	@OpenCombine.Published private(set) var buttonAccessibilityIdentifier: String?
+	@OpenCombine.Published private(set) var infoButtonAccessibilityIdentifier: String?
+	@OpenCombine.Published private(set) var deleteButtonAccessibilityIdentifier: String?
+	
+	@OpenCombine.Published private(set) var titleAccessibilityLabel: String?
+	@OpenCombine.Published private(set) var subtitleAccessibilityLabel: String?
+	@OpenCombine.Published private(set) var descriptionAccessibilityLabel: String?
+	@OpenCombine.Published private(set) var assetAccessibilityLabel: String?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:41]:
```suggestion
	@OpenCombine.Published private(set) var imageAccessibilityLabel: String?
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 11]
```diff
@@ -1422,7 +1422,17 @@ enum AppStrings {
 		enum Card {
 			static let fromNationWide = NSLocalizedString("Statistics_Card_From_Nationwide", comment: "")
 			static let fromCWA = NSLocalizedString("Statistics_Card_From_CWA", comment: "")
+			
+			enum LinkCardCommon {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1426]:
```suggestion
			enum LinkCard {
```

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 12]
```diff
@@ -486,6 +486,19 @@ enum AccessibilityIdentifiers {
 		static let modifyLocalIncidencesButton = "AppStrings.LocalStatistics.modifyLocalIncidencesButton"
 		static let modifyLocalIncidenceLabel = "AppStrings.LocalStatistics.modifyLocalIncidenceLabel"
 	}
+	
+	enum LinkCard {
+		enum PandemicRadar {
+			static let titleLabel = "AppStrings.LinkCard.PandemicRadar.titleLabel"
+			static let subtitleLabel = "AppStrings.LinkCard.PandemicRadar.subtitleLabel"
+			static let descriptionLabel = "AppStrings.LinkCard.PandemicRadar.descriptionLabel"
+			static let asset = "AppStrings.LinkCard.PandemicRadar.asset"
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:495]:
```suggestion
			static let image = "AppStrings.LinkCard.PandemicRadar.image"
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 13]
```diff
@@ -1530,6 +1540,8 @@ enum AppStrings {
 		enum Info {
 			static let title = NSLocalizedString("Statistics_Info_Title", comment: "")
 			static let subtitle = NSLocalizedString("Statistics_Info_Subtitle", comment: "")
+			static let pandemicRadarTitle = NSLocalizedString("Statistics_Info_PandemicRadar_Title", comment: "")
+			static let pandemicRadarText = NSLocalizedString("Statistics_Info_PandemicRadar_Text", comment: "")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1544]:
```suggestion
			static let pandemicRadarDescription = NSLocalizedString("Statistics_Info_PandemicRadar_Text", comment: "")
```

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 14]
```diff
@@ -2209,6 +2223,10 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 
 "Statistics_Info_Subtitle" = "Erklärung der Statistiken";
 
+"Statistics_Info_PandemicRadar_Title" = "Pandemieradar";
+
+"Statistics_Info_PandemicRadar_Text" = "Diverse Statistiken zur Corona-Pandemie, die auf einer Webseite des Robert Koch-Instituts bereitgestellt werden.\nBitte beachten Sie, dass die anfallenden Daten auf Ihr Datenvolumen angerechnet werden und die Webseite nicht barrierefrei ist.";
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2228]:
```suggestion
"Statistics_Info_PandemicRadar_Description" = "Diverse Statistiken zur Corona-Pandemie, die auf einer Webseite des Robert Koch-Instituts bereitgestellt werden.\nBitte beachten Sie, dass die anfallenden Daten auf Ihr Datenvolumen angerechnet werden und die Webseite nicht barrierefrei ist.";
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+	}
+	
+	@IBAction private func buttonTapped(_ sender: ENAButton) {
+		guard let url = viewModel?.buttonURL else {
+			Log.info("Link Card URL was not given.", log: .localStatistics)
+			return
+		}
+		onButtonTap?(url)
+	}
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:148]:
```suggestion
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+		viewModel?.$titleAccessibilityIdentifier
+			.assign(to: \.accessibilityIdentifier, on: titleLabel)
+			.store(in: &subscriptions)
+		viewModel?.$subtitleAccessibilityIdentifier
+			.assign(to: \.accessibilityIdentifier, on: subtitleLabel)
+			.store(in: &subscriptions)
+		viewModel?.$descriptionAccessibilityIdentifier
+			.assign(to: \.accessibilityIdentifier, on: descriptionLabel)
+			.store(in: &subscriptions)
+		viewModel?.$assetAccessibilityIdentifier
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:176]:
```suggestion
		viewModel?.$imageAccessibilityIdentifier
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+			.assign(to: \.accessibilityIdentifier, on: assetImageView)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:177]:
```suggestion
			.assign(to: \.accessibilityIdentifier, on: imageView)
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+		viewModel?.$titleAccessibilityLabel
+			.assign(to: \.accessibilityLabel, on: titleLabel)
+			.store(in: &subscriptions)
+		viewModel?.$subtitleAccessibilityLabel
+			.assign(to: \.accessibilityLabel, on: subtitleLabel)
+			.store(in: &subscriptions)
+		viewModel?.$descriptionAccessibilityLabel
+			.assign(to: \.accessibilityLabel, on: descriptionLabel)
+			.store(in: &subscriptions)
+		viewModel?.$assetAccessibilityLabel
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:198]:
```suggestion
		viewModel?.$imageAccessibilityLabel
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 19]
```diff
@@ -0,0 +1,219 @@
...
+			.assign(to: \.accessibilityLabel, on: assetImageView)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:199]:
```suggestion
			.assign(to: \.accessibilityLabel, on: imageView)
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift [hunk 20]
```diff
@@ -0,0 +1,219 @@
...
+			.assign(to: \.accessibilityLabel, on: infoButton)
+			.store(in: &subscriptions)
+		viewModel?.$deleteButtonAccessibilityLabel
+			.assign(to: \.accessibilityLabel, on: deleteButton)
+			.store(in: &subscriptions)
+		
+		titleLabel.accessibilityTraits = .header
+		subtitleLabel.accessibilityTraits = .staticText
+		descriptionLabel.accessibilityTraits = .staticText
+		assetImageView.accessibilityTraits = .image
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:214]:
```suggestion
		imageView.accessibilityTraits = .image
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+class HomeLinkCardViewModel {
+	
+	// MARK: - Init
+
+	init(for linkCard: SAP_Internal_Stats_LinkCard) {
+		switch HomeLinkCard(rawValue: linkCard.header.cardID) {
+		case .pandemicRadar:
+			setupPandemicRadar(for: linkCard)
+		case .none:
+			Log.info("Link card ID \(linkCard.header.cardID) is not supported", log: .ui)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:17]:
```suggestion
			Log.info("Card ID \(linkCard.header.cardID) is not supported", log: .ui)
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+		if #available(iOS 13.0, *) {
+			textAttachment.image = UIImage(
+				systemName: "rectangle.portrait.and.arrow.right",
+				withConfiguration: UIImage.SymbolConfiguration(weight: .semibold)
+			)?.withTintColor(
+				.enaColor(for: .buttonPrimary),
+				renderingMode: .alwaysOriginal
+			)
+		} else {
+			// Fallback iOS 12
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:59]: Adding a to.do so it is not forgotten

```suggestion
                        // to.do 
			// Fallback iOS 12
```

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2048]: This is the space between the button title attributed text and the text attachment (the sf symbol)

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+		textString.append(imageString)
+		
+		return textString
+	}()
+	
+	private func setupPandemicRadar(for linkCard: SAP_Internal_Stats_LinkCard) {
+		title = AppStrings.Statistics.Card.PandemicRadar.title
+		subtitle = AppStrings.Statistics.Card.PandemicRadar.subtitle
+		description = AppStrings.Statistics.Card.PandemicRadar.description
+		asset = UIImage(named: "Illu_Radar")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:80]:
```suggestion
		image = UIImage(named: "Illu_Radar")
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+		buttonTitle = buttonTitleAttributedString
+		buttonURL = URL(string: linkCard.url)
+		
+		titleAccessibilityIdentifier = AccessibilityIdentifiers.LinkCard.PandemicRadar.titleLabel
+		subtitleAccessibilityIdentifier = AccessibilityIdentifiers.LinkCard.PandemicRadar.subtitleLabel
+		descriptionAccessibilityIdentifier = AccessibilityIdentifiers.LinkCard.PandemicRadar.descriptionLabel
+		assetAccessibilityIdentifier = AccessibilityIdentifiers.LinkCard.PandemicRadar.asset
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:87]:
```suggestion
		imageAccessibilityIdentifier = AccessibilityIdentifiers.LinkCard.PandemicRadar.asset
```

File: src/xcode/ENA/ENA/Source/Scenes/StatisticsInfo/StatisticsInfoViewController.swift [hunk 25]
```diff
@@ -95,6 +95,15 @@ class StatisticsInfoViewController: DynamicTableViewController {
 				),
 				footer: .space(height: 16),
 				cells: [
+					// Pandemieradar
+					.title2(
+						text: AppStrings.Statistics.Info.pandemicRadarTitle,
+						accessibilityIdentifier: nil
+					) { _, cell, _ in cell.accessibilityTraits = .header },
+					.body(
+						text: AppStrings.Statistics.Info.pandemicRadarText,
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/StatisticsInfo/StatisticsInfoViewController.swift:104]:
```suggestion
						text: AppStrings.Statistics.Info.pandemicRadarDescription,
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift [hunk 26]
```diff
@@ -0,0 +1,181 @@
...
+		)
+	}
+	
+	func testAssetAccessibilityIdentifier_pandemicRadar() {
+		// Given
+		let sut = HomeLinkCardViewModel(for: pandemicRadarLinkCard)
+		
+		// Then
+		XCTAssertEqual(
+			sut.assetAccessibilityIdentifier,
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:144]:
```suggestion
			sut.imageAccessibilityIdentifier,
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift [hunk 27]
```diff
@@ -0,0 +1,181 @@
...
+			AccessibilityIdentifiers.LinkCard.PandemicRadar.asset
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:145]:
```suggestion
			AccessibilityIdentifiers.LinkCard.PandemicRadar.image
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift [hunk 28]
```diff
@@ -0,0 +1,181 @@
...
+		guard let buttonTitle = sut.buttonTitle else {
+			return XCTFail("Expect `buttonTitle`.")
+		}
+		
+		// Then
+		var textRange = NSRange(
+			location: 0,
+			length: AppStrings.Statistics.Card.LinkCardCommon.buttonTitle.count
+		)
+		let attachementRange = NSRange(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:59]:
```suggestion
		let attachmentRange = NSRange(
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCard.swift (hunk 5 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCard.swift:7]: Yes, for that we need to have a refactoring

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 29]
```diff
@@ -227,6 +233,45 @@ class HomeStatisticsTableViewCell: UITableViewCell {
 			)
 		}
 	}
+	
+	private func configurePandemicRadarCard(
+		onInfoButtonTap: @escaping CompletionVoid,
+		onAccessibilityFocus: @escaping CompletionVoid,
+		onUpdate: @escaping CompletionVoid
+	) {
+		guard let pandemicLinkCard = cellModel?.linkCards.first(where: { $0.header.cardID == 12 }) else {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:242]: Should this card be passed to the configure method and you check this logic before calling the function?

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2069]: Okay :)

File: src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift (hunk 4 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Extensions/NSAttributedString+Image.swift:9]: no, it's an inner property

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 6 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:120]: okay, done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 7 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:116]: I will leave a comment

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 8 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:23]: Setter is private, Getter is internal. I would keep it under internal, cause you can see it from outside

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2048]: @flxschmidt Please add the space with code then as the translation delivery doesn't necessarily take care of spaces.

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 9 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:33]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 10 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:41]: done

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 11 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1426]: okay

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift (hunk 12 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:495]: done

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 13 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1544]: "Text" is equal to the former ones. Would keep it equally

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 15 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:148]: Eagle Eye :)

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 17 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:177]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 18 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:198]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 19 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:199]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift (hunk 20 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardView.swift:214]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 22 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:59]: good point

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 23 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:80]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift (hunk 24 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeLinkCardView/HomeLinkCardViewModel.swift:87]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift (hunk 26 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:144]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift (hunk 28 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:59]: same

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift [hunk 30]
```diff
@@ -0,0 +1,181 @@
...
+			location: textRange.length,
+			length: buttonTitle.length
+		)
+		
+		XCTAssertEqual(
+			buttonTitle.attributedSubstring(from: textRange).string,
+			AppStrings.Statistics.Card.LinkCardCommon.buttonTitle
+		)
+
+		XCTAssertTrue(buttonTitle.containsAttachments(in: attachementRange))
```
[the remark this answers is not in the document]
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/HomeLinkCardViewModelTests.swift:69]: same

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift (hunk 29 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:242]: Could be part of the refactoring

**Review by puneetmahali1 — APPROVED** [review]

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 31]
```diff
@@ -2041,8 +2041,11 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 /* - General Card Texts */
 
 "Statistics_Card_From_Nationwide" = "Bundesweit";
+
 "Statistics_Card_From_CWA" = "über die Corona-Warn-App";
 
+"Statistics_Card_LinkCard_buttonTitle" = "Externen Link öffnen  ";
```
[the remark this answers is not in the document]
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: removed

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift (hunk 29 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:242]: @flxschmidt please comment to.do, so we don't forget?

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift (hunk 29 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:246]: we don't need this. When we want to have a complete generic solution, based on card sequence array, you would go through the sequence and map the types of cards ...

**Pull request merged**